src/libraries/Microsoft.PowerFx.Connectors/Execution/HttpFunctionInvoker.cs

```diff
@@ -123,6 +125,18 @@ public async Task<HttpRequestMessage> BuildRequest(FormulaValue[] args, IConvert
             }
 
             var url = (OpenApiParser.GetServer(_function.Servers, _httpClient) ?? string.Empty) + path + query.ToString();
+
+            // Replace connectionId or other parameters in the URL
+            foreach (Match m in new Regex(@"{(?<p>[^{}]+)}").Matches(url))
```

Comment:
> new Regex(@"{(?<p>[^{}]+)}").Matches(url)) [](http://example.com/codeflow?start=31&length=43)

running a regex at this layer on every invoke seems scary. 

Du we really need this? Especially for CDP case if we're already building URIs directly

Reply:
This Regex is very simple but I agree it's on the critical path.
We could cache the results (not in Match format of course) so that we would accelerate this a little bit.
We could also have a loop on url.Split('{')

Could be in a later PR or this one, I let you decide

Reply:
It's not just performance; it's risk and and suggests a layering issue and we need to convince ourselves it's safe.  

What about this PR is forcing us to add this? It's for building a request  and CDPTabularService builds the request directly (and I think it skips this method altogether?)

Reply:
It was a bug + I need it for this PR
We need these replacements for url parameters (like server, database) and I included ConnectionId